packages/grafana-data/src/transformations/matchers/nameMatcher.ts

```diff
@@ -4,32 +4,66 @@ import { FieldMatcherInfo, FrameMatcherInfo } from '../../types/transformations'
 import { stringToJsRegex } from '../../text/string';
 import { getFieldDisplayName } from '../../field/fieldState';
 
-// General Field matcher
-const fieldNameMacher: FieldMatcherInfo<string> = {
+export interface FieldNameMatcherOptions {
+  pattern?: string;
+  names?: string[];
+}
+
+const fieldNameMatcher: FieldMatcherInfo<FieldNameMatcherOptions> = {
   id: FieldMatcherID.byName,
   name: 'Field Name',
   description: 'match the field name',
-  defaultOptions: '/.*/',
+  defaultOptions: {
+    pattern: '/.*/',
+    names: [],
+  },
 
-  get: (pattern: string) => {
-    let regex = new RegExp('');
-    try {
-      regex = stringToJsRegex(pattern);
-    } catch (e) {
-      console.error(e);
-    }
-    return (field: Field) => {
-      return regex.test(getFieldDisplayName(field) ?? '');
+  get: (options: FieldNameMatcherOptions) => {
+    const regex = patternToRegex(options.pattern);
+    const matchByName = namesToRecord(options.names);
+
+    return (field: Field, frame: DataFrame, allFrames: DataFrame[]) => {
```

Comment:
Ignore this comment for this PR, but I've noticed that this API `field: Field, frame: DataFrame, allFrames: DataFrame[]` is something that is spreading in our codebase.

Reply:
Yeah, agree! Feels like there is some structure that we are missing that would solve this.

Reply:
maybe this? https://github.com/grafana/grafana/pull/24199 :)  but for these filtering functions not sure, only alternative would be to have some field display name set in some other more static way, when we build a full data frame list, but that feels also a bit backwards. tricky stuff

Reply:
Oh, I need to dig into that one once this release is shipped 👍